engine/access/rest/accounts.go

```diff
@@ -29,5 +46,46 @@ func getAccount(r *requestDecorator, backend access.API, link LinkGenerator) (in
 		}
 	}
 
-	return accountResponse(account), nil
+	expandContracts := r.expands(expandableFieldContracts)
+	expandKeys := r.expands(expandableFieldKeys)
+	return accountResponse(account, link, expandKeys, expandContracts)
+}
+
+func accountResponse(flowAccount *flow.Account, link LinkGenerator, expandKeys bool, expandContracts bool) (generated.Account, error) {
+
+	account := generated.Account{
+		Address: flowAccount.Address.String(),
+		Balance: int32(flowAccount.Balance),
+	}
+
+	if expandKeys {
+		account.Keys = accountKeysResponse(flowAccount.Keys)
+	} else {
+		account.Expandable = &generated.AccountExpandable{
+			Keys: expandableFieldKeys,
+		}
+	}
+
+	if expandContracts {
+		contracts := make(map[string]string, len(flowAccount.Contracts))
+		for name, code := range flowAccount.Contracts {
+			contracts[name] = toBase64(code)
+		}
+		account.Contracts = contracts
+	} else {
```

Comment:
I think the expandable can always be static (maybe we should check with @smnzhu) no matter what the query is. If that is true then we don't need to dynamically set them and they can just be as part of response generator.